### PR TITLE
fcrdns: add tests and improved localhost detection

### PR DESCRIPTION
**Pull request by msimerson**

### CODE REVIEW

**Comment by jaredj** [comment]
Is the [documentation](https://github.com/msimerson/qpsmtpd/blob/fcrdns/plugins/fcrdns#L9-L11) on this outdated?

>Determine if the SMTP sender has matching forward and reverse DNS.
>Sets the connection note fcrdns.

... but all I see is some karma adjustment and no note set in this file (before and after this PR)

File: t/plugin_tests/fcrdns [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+
+sub _is_valid_localhost {
+    my $self = shift;
+    my @passes = qw/ 127.0.0.1 127.1.1.1 127.255.255.255 /;
+    my @fails  = qw/ 128.0.0.1 126.1.1.1 /;
+
+    foreach my $pass (@passes) {
+        $self->qp->connection->remote_ip($pass);
+        cmp_ok( $self->is_valid_localhost(), '==', 1, "$pass, true");
```
Comment (jaredj) [inline t/plugin_tests/fcrdns]: petty, wouldn't `is( ..., 1 )` be just as well as `cmp_ok( ..., '==', 1 )` here, but a bit less to read?

Comment (msimerson) [inline t/plugin_tests/fcrdns]: :-1:

**Comment by msimerson** [comment]
Docs updated.

File: t/plugin_tests/fcrdns (hunk 1 above)
Comment (jaredj) [inline t/plugin_tests/fcrdns]: :question:

File: plugins/fcrdns [hunk 2]
```diff
@@ -168,35 +167,35 @@ sub is_valid_localhost {
...
     my $rh = $self->qp->connection->remote_host;
-    if ($rh && lc $self->qp->connection->remote_host eq 'localhost') {
-        $self->log(LOGDEBUG, "pass, remote_host is localhost");
-        return 1;
-    };
+    return 0 if ! $rh;
+    return 0 if lc $self->qp->connection->remote_host ne 'localhost';
 
     $self->adjust_karma(-1);
     $self->log(LOGINFO, "fail, not localhost");
```
Comment (jaredj) [inline plugins/fcrdns]: Are we subtracting karma from all remote hosts that are not localhost?  Or am I reading that wrong?

File: t/plugin_tests/fcrdns (hunk 1 above)
Comment (msimerson) [inline t/plugin_tests/fcrdns]: I agree your suggestion is petty and I prefer cmp_ok.

File: plugins/fcrdns (hunk 2 above)
Comment (msimerson) [inline plugins/fcrdns]: You are reading that wrong. We subtract karma from hosts that claim to be localhost (i.e., they have their rDNS set to 'localhost') but are not arriving from a loopback IP.

**Pull request merged**